mono/tests/test-runner.cs

```diff
@@ -611,4 +608,119 @@ class TestInfo {
 		// Char ::= #x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]	/* any Unicode character, excluding the surrogate blocks, FFFE, and FFFF. */
 		return Regex.Replace (text, @"[^\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFD\u10000-\u10FFFF]", "");
 	}
+
+	static void TryThreadDump (int pid, ProcessData data)
+	{
+		try {
+			TryGDB (pid, data);
+			return;
+		} catch (Exception) {
+		}
+
+#if !FULL_AOT_DESKTOP && !MOBILE
+		/* LLDB cannot produce managed stacktraces for all the threads */
+		try {
+			Syscall.kill (pid, Signum.SIGQUIT);
+			Thread.Sleep (1000);
+		} catch {
+		}
+#endif
+
+		try {
+			TryLLDB (pid, data);
+			return;
+		} catch (Exception) {
+		}
+	}
+
+	static void TryLLDB (int pid, ProcessData data)
+	{
+		string filename = Path.GetTempFileName ();
+
+		using (StreamWriter sw = new StreamWriter (new FileStream (filename, FileMode.Open, FileAccess.Write)))
+		{
+			sw.WriteLine ("process attach --pid " + pid);
+			sw.WriteLine ("thread list");
+			sw.WriteLine ("thread backtrace all");
+			sw.WriteLine ("detach");
+			sw.WriteLine ("quit");
+			sw.Flush ();
+
+			ProcessStartInfo psi = new ProcessStartInfo {
+				FileName = "lldb",
+				Arguments = "--batch --source " + filename + " --no-lldbinit",
```

Comment:
I assume filename could contain spaces e.g. on Windows so maybe better to enclose in quotes.